qemu/tests/pci_bridge_hotplug_unplug.py

```diff
@@ -0,0 +1,140 @@
+    if image_num + 1 != len(monitor_info_block.keys()):
+        test.fail("The amount of block devices in 'info block' is incorrect!")
+    logging.info("Check 'info block' succeed!")
+
+    error_context.context("Read and write data on all hotplugged disks!", logging.info)
+    sub_test_type = params.get("sub_test_type", "dd_test")
+    images = params.get("images")
+    images = images.split()
+    images.pop(0)
+    for image in images:
+        params["dd_if"] = "ZERO"
+        params["dd_of"] = image
+        utils_test.run_virt_sub_test(test, params, env, sub_test_type)
+
+        params["dd_if"] = image
+        params["dd_of"] = "NULL"
+        utils_test.run_virt_sub_test(test, params, env, sub_test_type)
+    logging.info("Read and write data on all hotplugged disks succeed!")
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    [pci-bridge] Hotplug and unplug different devices to 1 pci-bridge
+    [pci-bridge] Hotplug and unplug different devices to different pci-bridges
+    This case will:
+    1) Start the guest.
+    2) Hotplug different devices to pci-bridge(s).
+    3) Check devices.
+    4) Hot-unplug devices.
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    pci_bridge_num = int(params.get("pci_bridge_num", 0))
+    if pci_bridge_num > 1:
+        raise NotImplementedError()
+
+    error_context.context("Get the main VM", logging.info)
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+
+    login_timeout = int(params.get("login_timeout", 360))
+    session = vm.wait_for_login(timeout=login_timeout)
+
+    image_list = params.get("images").split()
+    image_num = int(params.get("image_num"))
+    if len(image_list) < image_num + 1:
+        test.fail("No sufficient pre-prepared images!")
+
+    device_list = []
+    if pci_bridge_num == 0:
+        test.fail("pci-bridge number is not specified!")
+    elif pci_bridge_num == 1:
+        pci_bridge_id = params.get("pci_controllers").strip()
+
+        for fmt in ('virtio', 'scsi-hd', 'usb3'):
+            error_context.context("Hotplug a %s disk!" % fmt, logging.info)
+            device_list += disk_hotplug(test, vm, params, image_list.pop(), fmt, pci_bridge_id)
+            logging.info("Hotplug a %s disk succeed!" % fmt)
+        time.sleep(60)
+
+        error_context.context("Check all hotplugged disks!", logging.info)
+        check_all_hotplugged_disks(test, params, env)
+        logging.info("Check all hotplugged disks succeed!")
+
+    else:
+        raise NotImplementedError()
+
+    error_context.context("Unplug those hotplugged devices!", logging.info)
+    device_list.reverse()
+    for dev in device_list:
+        dev.unplug(vm.monitor)
+        dev.verify_unplug("", vm.monitor)
+        logging.info("Unplug %s succeed!" % dev)
+    logging.info("Unplug all hotplugged devices succeed!")
```

Comment:
IMHO, it will be better if we replace all "succeed" with "successfully" regarding the grammar. Thanks